meta-intel-iot-security/meta-security-smack/recipes-core/systemd/systemd/0001-core-smack-add-support-for-defining-smack-rules-path-v229.patch

```diff
@@ -0,0 +1,211 @@
+Date: Wed, 16 Mar 2016 15:12:09 +0200
+Subject: [PATCH] core/smack: add support for defining smack rules path in
+ configure
+
+This patch is made for enabling smack to read its rules also from
+somewhere else than the hard coded /etc. The goal is to enable
+some variant of "stateless" distro where things should work at
+least somehow even if /etc is empty. In this case it is possible
+to configure the smack rules in the default dir, /etc or both.
+
+Upstream-Status: Pending
+
+Signed-off-by: Jaska Uimonen <jaska.uimonen@intel.com>
+---
+ configure.ac           |  16 +++++++
+ src/core/smack-setup.c | 114 +++++++++++++++++++++++++++++++------------------
+ 2 files changed, 88 insertions(+), 42 deletions(-)
+
+diff --git a/configure.ac b/configure.ac
+index 5fd73c5..73e6fe6 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -694,6 +694,21 @@ AS_HELP_STRING([--with-smack-default-process-label=STRING],
+         [AC_DEFINE_UNQUOTED(SMACK_DEFAULT_PROCESS_LABEL, ["$withval"], [Default SMACK label for executed processes])],
+         [])
+ 
++AC_ARG_WITH(smack-default-rules-path,
++AS_HELP_STRING([--with-smack-default-rules-path=STRING],
++        [default path for SMACK rules]),
++        [DEFAULT_SMACK_RULES_PATH="$withval"],
++        [DEFAULT_SMACK_RULES_PATH="/etc"])
++
++AC_DEFINE_UNQUOTED(DEFAULT_SMACK_RULES_PATH, [$DEFAULT_SMACK_RULES_PATH], [Default path for SMACK rules])
++AC_SUBST(DEFAULT_SMACK_RULES_PATH)
++
++if test "x${have_smack}" = xyes ; then
++        AC_DEFINE(HAVE_SMACK, 1, [Define if SMACK is available])
++fi
++
++AM_CONDITIONAL([HAVE_SMACK], [test "x$have_smack" = "xyes"])
++
+ # ------------------------------------------------------------------------------
+ AC_ARG_ENABLE([gcrypt],
+         AS_HELP_STRING([--disable-gcrypt],[Disable optional GCRYPT support]),
+@@ -1607,6 +1622,7 @@ AC_MSG_RESULT([
+         Maximum System UID:      ${SYSTEM_UID_MAX}
+         Maximum System GID:      ${SYSTEM_GID_MAX}
+         Certificate root:        ${CERTIFICATEROOT}
++        Smack rules path:        ${DEFAULT_SMACK_RULES_PATH}
+ 
+         CFLAGS:                  ${OUR_CFLAGS} ${CFLAGS}
+         CPPFLAGS:                ${OUR_CPPFLAGS} ${CPPFLAGS}
+diff --git a/src/core/smack-setup.c b/src/core/smack-setup.c
+index 0c26e85..f1ad970 100644
+--- a/src/core/smack-setup.c
++++ b/src/core/smack-setup.c
+@@ -35,9 +35,22 @@
+ #include "smack-setup.h"
+ #include "string-util.h"
+ #include "util.h"
++#include "strv.h"
+ 
+ #ifdef HAVE_SMACK
+ 
++#ifndef STRINGIFY
++#  define STRINGIFY(foo) #foo
++#endif
++
++#define DEFAULT_RULES_PATH_SMACK STRINGIFY(DEFAULT_SMACK_RULES_PATH) "/smack/accesses.d/"
++#define DEFAULT_RULES_PATH_CIPSO STRINGIFY(DEFAULT_SMACK_RULES_PATH) "/smack/cipso.d/"
++#define DEFAULT_RULES_PATH_NETLABEL STRINGIFY(DEFAULT_SMACK_RULES_PATH) "/smack/netlabel.d/"
++
++#define RULES_PATH_SMACK "/etc/smack/accesses.d/"
++#define RULES_PATH_CIPSO "/etc/smack/cipso.d/"
++#define RULES_PATH_NETLABEL "/etc/smack/netlabel.d/"
```

Comment:
Perhaps change RULES_PATH_SMACK to ETC_RULES_PATH_SMACK, to make it clear that this the hard-coded directory in /etc.

I think upstream had concerns about defines for string constants, so expect some comments along those lines when submitting upstream. There might also be some trick to avoid the STRINGIFY, but I don't have it readily available.